packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5900,3 +5900,33 @@ FlutterErrorDetails _debugReportException(
   FlutterError.reportError(details);
   return details;
 }
+
+/// Object used by [MultiChildRenderObjectElement] for the [Element.slot] value.
+///
+/// A slot for a [MultiChildRenderObjectElement] consists of an [index]
+/// identifying where the child occupying this slot is located in the
+/// [MultiChildRenderObjectElement]'s child list and an arbitrary [value] that
+/// can further define where the child occupying this slot fits in its
+/// parent's child list.
+@immutable
+class IndexedSlot<T> {
+  /// Creates an [IndexedSlot] with the provided [index] and slot [value].
+  const IndexedSlot(this.index, this.value);
+
+  /// Information to define where the child occupying this slot fits in its
+  /// parent's child list.
+  final T value;
+
+  /// The index of this slot in the parent's child list.
+  final int index;
```

Comment:
Now that I look at this again I wonder whether we should just take a generic salt in here instead of an (explicit) index. In either case it probably also makes sense to remove the public getter for that value since nobody should really depend on what the actual value is.